ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/utils/ThreadUtil.java

```diff
@@ -28,7 +31,7 @@
     /**
      * The ID associated with the thread.
      */
-    private static final ThreadLocal THREAD_ID = new ThreadLocal() ;
+    private static final Map<Thread,String> THREAD_ID = new HashMap<Thread,String>() ;
```

Comment:
I wondered if you meant that but then I thought that catting the "TSThread"+getId() each time would be more expensive than storing a string in the map?